net/src/sim2h_worker.rs

```diff
@@ -0,0 +1,374 @@
+//! provides worker that makes use of sim2h
+
+use crate::connection::{
+    net_connection::{NetHandler, NetWorker},
+    NetResult,
+};
+use holochain_json_api::{error::JsonError, json::JsonString};
+use lib3h_protocol::{protocol_client::Lib3hClientProtocol, protocol_server::Lib3hServerProtocol, Address};
+use log::*;
+use sim2h::WireMessage;
+use url::Url;
+use lib3h::transport::websocket::actor::{GhostTransportWebsocket};
+use lib3h::transport::websocket::tls::TlsConfig;
+use lib3h::transport::protocol::{
+    RequestToChild, RequestToChildResponse, RequestToParent, TransportActorParentWrapper
+};
+use holochain_tracing::Span;
+use lib3h_zombie_actor::{GhostParentWrapper, GhostCallbackData, WorkWasDone};
+use lib3h_zombie_actor::GhostCanTrack;
+use failure::err_msg;
+use lib3h_protocol::protocol::*;
+use lib3h_protocol::data_types::{GenericResultData, Opaque};
+use lib3h_protocol::uri::Lib3hUri;
+use std::convert::TryFrom;
+use detach::Detach;
+
+#[derive(Deserialize, Serialize, Clone, Debug, DefaultJson, PartialEq)]
+pub struct Sim2hConfig {
+    pub sim2h_url: String,
+}
+
+/// removed lifetime parameter because compiler says ghost engine needs lifetime that could live statically
+#[allow(non_snake_case, dead_code)]
+pub struct Sim2hWorker {
+    handler: NetHandler,
+    transport: Detach<TransportActorParentWrapper<Sim2hWorker, GhostTransportWebsocket>>,
+    inbox: Vec<Lib3hClientProtocol>,
+    to_core: Vec<Lib3hServerProtocol>,
+    num_ticks: u32,
+    server_url: Lib3hUri,
+}
+
+impl Sim2hWorker {
+    pub fn advertise(self) -> url::Url {
+        Url::parse("ws://example.com").unwrap()
+    }
+
+    /// Create a new worker connected to the sim2h instance
+    pub fn new(handler: NetHandler, config: Sim2hConfig) -> NetResult<Self> {
+        let transport_raw = GhostTransportWebsocket::new(
+            Address::from("sim2h-worker-transport"),
+            TlsConfig::Unencrypted,
+            Address::from("sim2h-network"),
```

Comment:
personally I know that these are hardcoded to match settings over in sim2h, but that might deserve some code comments

Reply:
magic strings and all

Reply:
Those strings are not used at all, definitely not in sim2h. I’m not completely clear on what the websocket ghost actor does with them. I had set them like this to give a hint should they appear somewhere when debugging.

Reply:
I guess I just meant these values match these values over in sim2h and I got the impression things wouldn't work if those values didn't match. Could be very easily wrong tho. https://github.com/holochain/sim2h/blob/b2453a222830c2dcf2d2f8b3bba5acc59f85d65b/crates/sim2h_server/src/main.rs#L24